matrix/src/lib.rs

```diff
@@ -174,25 +177,42 @@ pub trait Matrix<T: Send + Sync>: Send + Sync {
     }
 
     /// Compute Mᵀv, aka premultiply this matrix by the given vector,
-    /// aka scale each row by the corresponding entry in `v` and take the row-wise sum.
+    /// aka scale each row by the corresponding entry in `v` and take the sum across rows.
     /// `v` can be a vector of extension elements.
     #[instrument(level = "debug", skip_all, fields(dims = %self.dimensions()))]
     fn columnwise_dot_product<EF>(&self, v: &[EF]) -> Vec<EF>
     where
         T: Field,
         EF: ExtensionField<T>,
     {
-        self.par_rows().zip(v).par_fold_reduce(
-            || EF::zero_vec(self.width()),
-            |mut acc, (row, &scale)| {
-                acc.iter_mut().zip(row).for_each(|(a, x)| *a += scale * x);
-                acc
-            },
-            |mut acc_l, acc_r| {
-                acc_l.iter_mut().zip(acc_r).for_each(|(l, r)| *l += r);
-                acc_l
-            },
-        )
+        let packed_width = self.width().next_multiple_of(T::Packing::WIDTH);
```

Comment:
I think we ideally want `self.width().div_ceil(T::Packing::WIDTH)` here, if this will be the number of packed elements in the accumulator?